invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js

```diff
@@ -423,6 +423,9 @@ class CommunityProfileForm extends Component {
     // Clean values
     submittedCommunity = removeEmptyValues(submittedCommunity);
 
+    // Remove ui object
+    delete submittedCommunity.ui;
```

Comment:
isn't it better to make the `.ui` key dump_only and let the backend handle it? it is going to be quite problematic to always remember to remove it in the frontend

Reply:
I am not 100% sure about this, I understand the point that it can be a bit more problematic but I think it should be the client responsible for sending some valid data and not the backend adapting to the client needs, note that the schema used to fetch the data is different than the one used to post the data, this is due to localization as we need to translate the vocabularies. That said we could easily add to the community Marshamallow schema `unknown = EXCLUDE` but this implies that when using the API we will not get an error if we add non-existing fields, I am fine with this, if this works for you I could add this change but want to double check with you first.

Please note that for the records we do also remove the `ui` object before sending the data  (see code [here](https://github.com/inveniosoftware/react-invenio-deposit/blob/master/src/lib/DepositRecordSerializer.js#L362-L370)). If we want to change this behaviour I think that we should do it for both. 

I am also available to discuss IRL if it's easier.

Reply:
@slint @alejandromumo any idea why the `unknown = EXCLUDE` was not added? Simply forgotten?

Reply:
Probably yes, it was forgotten... This was one of the very first forms we did in React, so the common services/resources/schemas practices we have now were not that common.